pkg/client/userd/trafficmgr/install.go

```diff
@@ -275,12 +273,59 @@ func useAutoInstall(podTpl *core.PodTemplateSpec) (bool, error) {
 	return !(webhookInjected || manuallyManaged), nil
 }
 
+type ServiceProps struct {
+	Service            *core.Service
+	ServicePort        *core.ServicePort
+	Container          *core.Container
+	ContainerPortIndex int
+}
+
+// ExploreSvc finds the matching service, its comtainters, and their ports
+func ExploreSvc(c context.Context, portNameOrNumber, svcName string, obj k8sapi.Workload) (*ServiceProps, error) {
```

Comment:
Unless there's a plan to later use this function from other packages, I suggest not exporting it. Use the name `exploreSvc` and return the type `*serviceProps`.